### PR TITLE
BZ-1285433 - WAS: creating a persistable data object causes an error

### PR DESCRIPTION
**Pull request by wmedvede**
BZ fixed and tests added.

### CODE REVIEW

**Comment by wmedvede** [comment]
@manstis 

Hello Mike, would you mind to review this PR.
PS: peer PR for 6.3.x apply but will be created when the green light to modify 6.3.x is on.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserverTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        when( descriptorPath.toURI() ).thenReturn( DESCRIPTOR_PATH );
+        when( descriptorService.calculatePersistenceDescriptorPath( any( Project.class ) ) ).thenReturn( descriptorPath );
+        when( descriptorService.load( descriptorPath ) ).thenReturn( descriptorModel );
+        when( ioService.exists( any( org.uberfire.java.nio.file.Path.class ) ) ).thenReturn( true );
+
+        DataObjectCreatedEvent createdEvent = new DataObjectCreatedEvent( project, dataObject );
+        eventObserver.onDataObjectCreated( createdEvent );
+
+        verify( descriptorService, times( 0 ) ).save( eq( descriptorPath ), eq( descriptorModel ), any( Metadata.class ), anyString() );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserverTest.java]: ```never()``` is an alternative to ```times( 0 )``` but no problem.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserverTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserverTest.java]: ```never()``` is an alternative to ```times( 0 )``` but no problem.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
master: 9d794675a6f40496e7550c5c1df72017ed07f4b0

**Pull request closed without merging**